### PR TITLE
Convert to Rust 2018

### PR DESCRIPTION
**Pull request by marmeladema**
Do we want to convert to Rust 2018 in the first place?

### CODE REVIEW

**Comment by RReverser** [comment]
What for? I tried in the past but I don't like all these `crate::` everywhere.

**Comment by marmeladema** [comment]
I wanted to use the `?` repetition operator in macro rules. But i will try to see if I can do what I want without it.

**Comment by RReverser** [comment]
That's actually a good point - if it already starts limiting our syntax usage, then it's not worth it and might be fine to switch.

File: engine/src/lib.rs [hunk 1]
```diff
@@ -55,26 +55,7 @@
...
-extern crate serde_json;
-
-extern crate cidr;
-extern crate fnv;
-extern crate indexmap;
-extern crate memmem;
-
-#[cfg(feature = "regex")]
-extern crate regex;
+use failure;
```
Comment (RReverser) [inline engine/src/lib.rs]: This shouldn't be necessary.

Comment (marmeladema) [reply engine/src/lib.rs]: Removed

**Comment by inikulin** [comment]
Interesting, how come master branch CI has failed even though it was green for the branch? Do we have different setup or something?

**Comment by marmeladema** [comment]
I might be wrong but I think it's a new clippy warning due to rust 1.34
We are not using a specific rust version, just the last stable one and 1.34 was released yesterday.

**Comment by RReverser** [comment]
@marmeladema Can you submit a fix please?

**Comment by marmeladema** [comment]
https://github.com/cloudflare/wirefilter/pull/18

**Pull request merged**